sparta/sparta/ports/Port.hpp

```diff
@@ -409,6 +409,9 @@ namespace sparta
          */
         void precedes(InPort & consumer)
         {
+            sparta_assert(getScheduleable_().getSchedulingPhase() == consumer.getScheduleable_().getSchedulingPhase(),
```

Comment:
Shouldn't this sort of behavior from the modeler be a candidate for ```std::logic_error``` exception?

Reply:
🤣 Yeah, we could argue that for most of the C++ exceptions in Sparta.  For now, I'm playing it safe ensuring the framework will catch this exception for the user and provide the standard error dump + backtrace...

Reply:
Fair enough.